pimd/pim6_cmd.c

```diff
@@ -33,16 +33,445 @@
+       "Detailed output\n"
+       "Name of interface or neighbor\n"
+       JSON_STR)
+{
+	struct vrf *vrf_struct;
+	json_object *json_parent = NULL;
+
+	vrf_struct = vrf_lookup_by_name(vrf ? vrf : VRF_DEFAULT_NAME);
+
+	if (!vrf_struct)
+		return CMD_WARNING;
+
+	if (json)
+		json_parent = json_object_new_object();
+	if (detail)
+		pim_show_neighbors_single(vrf_struct->info, vty, detail,
+					  json_parent);
+	else
+		pim_show_neighbors(vrf_struct->info, vty, json_parent);
+
+	if (json)
+		vty_json(vty, json_parent);
+
+	return CMD_SUCCESS;
+}
+
+DEFPY (show_ipv6_pim_neighbor_vrf_all,
+       show_ipv6_pim_neighbor_vrf_all_cmd,
+       "show ipv6 pim vrf all neighbor [[detail|WORD]$detail] [json$json]",
+       SHOW_STR
+       IPV6_STR
+       PIM_STR
+       VRF_CMD_HELP_STR
+       "PIM neighbor information\n"
+       "Detailed output\n"
+       "Name of interface or neighbor\n"
+       JSON_STR)
+{
+	struct vrf *vrf_struct;
+	json_object *json_parent = NULL;
+	json_object *json_vrf = NULL;
+
+	if (json)
+		json_parent = json_object_new_object();
+	RB_FOREACH (vrf_struct, vrf_name_head, &vrfs_by_name) {
+		if (!json)
+			vty_out(vty, "VRF: %s\n", vrf_struct->name);
+		else
+			json_vrf = json_object_new_object();
+		if (detail)
+			pim_show_neighbors_single(vrf_struct->info, vty, detail,
+						  json_vrf);
+		else
+			pim_show_neighbors(vrf_struct->info, vty, json_vrf);
+
+		if (json)
+			json_object_object_add(json_parent, vrf_struct->name,
+					       json_vrf);
+	}
+	if (json)
+		vty_json(vty, json_parent);
+
+	return CMD_SUCCESS;
+}
+
+DEFPY (show_ipv6_pim_nexthop,
+       show_ipv6_pim_nexthop_cmd,
+       "show ipv6 pim [vrf NAME] nexthop",
+       SHOW_STR
+       IPV6_STR
+       PIM_STR
+       VRF_CMD_HELP_STR
+       "PIM cached nexthop rpf information\n")
+{
+	struct vrf *vrf_struct;
+
+	vrf_struct = vrf_lookup_by_name(vrf ? vrf : VRF_DEFAULT_NAME);
+
+	if (!vrf_struct)
+		return CMD_WARNING;
+
+	pim_show_nexthop(vrf_struct->info, vty);
+
+	return CMD_SUCCESS;
+}
+
+DEFPY (show_ipv6_pim_nexthop_lookup,
+       show_ipv6_pim_nexthop_lookup_cmd,
+       "show ipv6 pim [vrf NAME] nexthop-lookup [X:X::X:X$source] [X:X::X:X$group]",
+       SHOW_STR
+       IPV6_STR
+       PIM_STR
+       VRF_CMD_HELP_STR
+       "PIM cached nexthop rpf lookup\n"
+       "Source/RP address\n"
+       "Multicast Group address\n")
+{
+	struct prefix nht_p;
+	int result = 0;
+	pim_addr src_addr, grp_addr;
+	pim_addr vif_source;
+	struct prefix grp;
+	struct pim_nexthop nexthop;
+	char nexthop_addr_str[PIM_ADDRSTRLEN];
+	struct vrf *vrf_struct;
+
+	vrf_struct = vrf_lookup_by_name(vrf ? vrf : VRF_DEFAULT_NAME);
+
+	if (!vrf_struct)
+		return CMD_WARNING;
+
+	memcpy(&src_addr, &source, sizeof(source));
+	memcpy(&grp_addr, &group, sizeof(group));
+	if (!IPV6_ADDR_SAME(&source, &in6addr_any))
+		memcpy(&src_addr, &source, sizeof(source));
+	if (!IPV6_ADDR_SAME(&group, &in6addr_any))
+		memcpy(&grp_addr, &group, sizeof(group));
+
```

Comment:
Same as above, Might be we will get comment to have a common wrapper for IPV4 and IPV6